magpy/gui/magpy_gui.py

```diff
@@ -1495,7 +1508,15 @@ def OnInitialPlot(self, stream, restore = False):
             self.plotoptlist.append(self.plotopt)
 
         self.plot_p.guiPlot([self.plotstream],[keylist], plotopt=self.plotopt)
-
+        boxes = ['x','y','z','f']
+        for box in boxes:
+            if box in self.shownkeylist:
+                exec('self.menu_p.str_page.'+box+'CheckBox.Enable()')
```

Comment:
Use `getattr` instead of `exec` to access the checkbox at the start of the loop, then use as needed:

```python
checkbox = getattr(self.menu_p.str_page, box + 'CheckBox')
```